### ISSUE TITLE
Not possible to enable SSL redirects with default SSL certificate

### ISSUE THREAD

**Issue by alanpearce** [jcmoraisjr/haproxy-ingress#384]
It doesn't seem to be possible to make HTTP->HTTPS redirects the default when using `default-ssl-certificate`, or at least it's not clear to me how to do it. I saw in https://github.com/jcmoraisjr/haproxy-ingress/issues/299#issuecomment-460943214 that `ssl-redirect` depends on having TLS configured on an ingress, which is not the case here as it's been set as the default.

Config (via helm chart 0.0.15):
```
controller:
  image:
    tag: "v0.7-beta.7"
  extraArgs:
    default-ssl-certificate: ingress/my-cert
    reload-strategy: reusesocket
  config:
    forwardfor: ifmissing
    ssl-redirect: "true"
```

The image tag is set to `v0.7-beta.7` as a workaround for #333. I tried with v0.7.2 in another environment and the behaviour is the same.

Example ingress:

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: foo
spec:
  rules:
  - host: foo.abc.xyz
    http:
      paths:
      - backend:
          serviceName: foo-service
          servicePort: http
        path: /
```

Adding the `ingress.kubernetes.io/ssl-redirect: "true"` annotation to the ingress doesn't seem to do anything either (we would want to avoid doing this for all out ingresses anyway).

I see
```
% http --headers foo.abc.xyz
HTTP/1.1 200 OK
Accept-Ranges: bytes
Cache-Control: public, s-maxage=86400, max-age=600, must-revalidate
Content-Length: 1721
Content-Type: text/html; charset=utf-8
Date: Mon, 09 Sep 2019 09:46:08 GMT
Last-Modified: Thu, 29 Aug 2019 14:15:40 GMT
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-Xss-Protection: 1; mode=block
```

When I would expect something like

```
% http --headers foo.abc.xyz
HTTP/1.1 301 Moved Permanently
Date: Mon, 09 Sep 2019 09:44:28 GMT
Location: https://foo.abc.xyz
```

**Comment by jcmoraisjr** [comment]
Hi, thanks for reporting. I'll need a few more days to understand what's happening here. In the mean time please check if `v0.8-beta.1` fixes both #333 and this one on your environment.

**Comment by alanpearce** [comment]
`v0.8-beta.1` didn't work, but I saw there's `v0.8-beta.2` and that seems to work well.
